### PR TITLE
some reference and move fixes

### PR DESCRIPTION
**Pull request by cyyever**
This PR introduces some modifications:
1. We find out some const function parameters that can be passed by reference and add the reference.
2. We find more opportunists of passing by value and change them accordingly. 
3. Some use-after-move errors are fixed.

cc @jgong5 @mingfeima @XiaobingSuper @sanchitintel @ashokei @jingxu10

### CODE REVIEW

File: aten/src/ATen/quantized/QTensorImpl.cpp [hunk 1]
```diff
@@ -3,22 +3,20 @@
 namespace at {
 
 QTensorImpl::QTensorImpl(
-    Storage&& storage,
+    Storage storage,
```
Comment (Skylion007) [inline aten/src/ATen/quantized/QTensorImpl.cpp:6]: Why change this from an rvalue?

Comment (cyyever) [reply aten/src/ATen/quantized/QTensorImpl.cpp:6]: Not sure it was a wrong optimisation or deliberately. It is better to revert the change.

**Review by Skylion007 — APPROVED** [review]
Please improve the PR description, otherwise this looks good.

**Pull request closed without merging**